sdk/appconfiguration/app-configuration/src/internal/helpers.ts

```diff
@@ -232,42 +231,40 @@ export function serializeAsConfigurationSettingParam(
 /**
  * @internal
  */
-export function transformKeyValueResponseWithStatusCode<
-  T extends KeyValue & HttpResponseField<any>
->(kvp: T): ConfigurationSetting & { eTag?: string } & HttpResponseField<any> & HttpResponseFields {
-  return normalizeResponse(kvp, <
-    ConfigurationSetting & HttpResponseField<any> & HttpResponseFields
-  >{
+export function transformKeyValueResponseWithStatusCode<T extends KeyValue>(
+  kvp: T,
+  status: number | undefined
+): ConfigurationSetting & { eTag?: string } & HttpResponseFields {
+  const response = {
     ...transformKeyValue(kvp),
-    statusCode: kvp._response.status,
-  });
+    statusCode: isDefined(status) ? status : -1,
```

Comment:
nullish coalescing operator was designed for this exact case :)

```suggestion
    statusCode: status ?? -1,
```

Reply:
I was thinking this, but wondered 0 could be a value and since dealing with numbers, went with isDefined.
But I agree, ?? is enough here.

Reply:
`??` will not replace `0` or `""` or anything that's not `null` or `undefined` - that's the main advantage over `||`